Correct some CPU selections in tools

kjbracey opened the pull request:
### Description

* For ARMC6, core types `Cortex-M4` and `Cortex-M7` did not explicitly add `--fpu=none`, so it defaulted to assuming FPU present. This would cause a compilation error if the target's cmsis.h had `__FPU_PRESENT` defined to 0.

* For GCC, `Cortex-M33FE` did not include `+dsp` in the architecture selection.

* For ARMC5 and ARMC6, `Cortex-M0+` did not pass `M0plus` to the non-Clang tools.

* Add missing `Cortex-M33E` DSP Extension without FP possibility.

* Switch KW24D to use ARMC6 - it was using ARMC5 because it has non-FP Cortex-M4 and hit this tool problem.

### Pull request type

    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

kjbracey committed:
Correct some CPU selections in tools

* For ARMC6, core types `Cortex-M4` and `Cortex-M7` did not explicitly
  add `--fpu=none`, so it defaulted to assuming FPU present. This would
  cause a compilation error if the target's cmsis.h had `__FPU_PRESENT`
  defined to 0.

* For GCC, `Cortex-M33FE` did not include `+dsp` in the architecture
  selection.

* For ARMC5 and ARMC6, `Cortex-M0+` did not pass `M0plus` to the
  non-Clang tools.

bridadan reviewed (approved):
Nice catch! Looks good to me, probably should run this past @deepikabhavnani too.

kjbracey committed:
Tools: Add "Cortex-M33E" option

There was a gap in our pattern - we didn't offer M33 with DSP Extension
but no floating-point.

kjbracey committed:
KW24D: Use default ARM compiler (ARMC6)

KW24D was set to ARMC5 because the ARMC6 tooling didn't correctly handle
Cortex-M4 without floating-point. Now fixed.
